sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -753,7 +762,13 @@ private[hive] class HiveClientImpl(
         assert(s.values.forall(_.nonEmpty), s"partition spec '$s' is invalid")
         s
     }
-    val parts = shim.getPartitions(client, hiveTable, partSpec.asJava).map(fromHivePartition)
+    val parentUri = if (hiveTable.getDataLocation.toUri.isAbsolute) {
+      None
+    } else {
+      convertHiveTableToCatalogTable(hiveTable).storage.locationUri
```

Comment:
shoudn't this be `CatalogUtils.stringToURI(client.getDatabase(h.getDbName).getLocationUri)`?

Reply:
+1